src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1147,9 +1147,32 @@ async ValueTask<NpgsqlDataReader> ExecuteReaderAsync(CommandBehavior behavior, b
                 if (conn.TryGetBoundConnector(out var connector))
                 {
                     connector.StartUserAction(this);
+
+                    CancellationTokenSource? cts = null;
+                    CancellationTokenRegistration? registration = null;
+                    var finalCt = CancellationToken.None;
+                    var cancellationTimeout = connector.Settings.CancellationTimeout;
```

Comment:
Move this inside the inner try block below as it's not used outside of it (and maybe just perform the multiplication on the same line?)